source/server/hot_restart_impl.cc

```diff
@@ -93,7 +93,7 @@ HotRestartImpl::HotRestartImpl(Options& options, Api::OsSysCalls& os_sys_calls)
       log_lock_(shmem_.log_lock_), access_log_lock_(shmem_.access_log_lock_),
       stat_lock_(shmem_.stat_lock_), init_lock_(shmem_.init_lock_) {
 
-  my_domain_socket_ = bindDomainSocket(options.restartEpoch());
+  my_domain_socket_ = bindDomainSocket(options.restartEpoch(), os_sys_calls);
```

Comment:
Hm, this works for this particular patch, but I think it's going to a pain to drill os_sys_calls to everything that needs it.
If we need much more of this we should probably consider intercepting system calls (I can try to dig up what we did for QUIC and see if it's open source) or something like an inject-able singleton.

Reply:
Yeah agreed we might need to do better here. If there is a better way of ultimately doing this definitely interested. Do you mind re-approving for the comment fix?